qa/tasks/vstart_runner.py

```diff
@@ -268,9 +268,19 @@ def _perform_checks_and_return_list_of_args(self, args, omit_sudo):
         # treated differently by Python's shell simulation. Only latter has
         # the desired effect.
         if 'sudo' in args and '-c' in args:
-            if args.index('-c') != len(args) - 2:
-                raise RuntimeError("All args after '-c' should be a single "
-                                   "string")
+            # make sure "-c" doesn't belong to some other command.
+            if 'python3' not in args or ('python3' in args and
+               (args.index('sudo') < args.index('-c') < args.index('python3'))):
+                index_of_cmd = args.index('-c') + 1
```

Comment:
This scares me with how complex and specific to python it is. Can you find a way to simplify it or maybe use argparse to look for the `-c` option?

Reply:
>  This scares me with how complex and specific to python it is.

It gets more complex when all python executables (`python`, `python2`, `python3`) will be checked for. My original intent was to make sure that nobody would have to spend time on trivial problems like these when writing tests. But after I wrote the following for testing PR #27805, I am wondering if I should give up on this unless I can simplify all the conditions here.

```
         if 'sudo' in args and '-c' in args:
            # make sure "-c" doesn't belong to some other command.
            pyexe_not_troublesome = False
            for pyexe in ['python', 'python2', 'python3']:
                if (pyexe not in args or (pyexe in args and
                   (args.index('sudo') < args.index('-c') < args.index(pyexe)))):
                    continue
                else:
                    break
            else:
                pyexe_not_troublesome = True

            if pyexe_not_troublesome:
                index_of_cmd = args.index('-c') + 1
                # finally, make sure that command to be executed as other user is
                # a single argument.
                if (args.index('-c') != len(args) - 2 and
                   (args[index_of_cmd][0] in ["'", '"'] and
                    args[index_of_cmd][-1] in ["'", '"'])):
                    raise RuntimeError("When using '-c' switch of sudo command, all "
                                       "args after it should be a single string.\n "
                                       "args - " + str(args))
```

I had not though about argparse, I will look into it.

Reply:
I've rewritten this part; let me know if you still think that there's scope for simplification.